src/lib/chat/matrix-client.ts

```diff
@@ -556,12 +610,13 @@ export class MatrixClient implements IChatClient {
     return this.getLatestEvent(room, EventType.RoomCreate)?.getTs() || 0;
   }
 
-  private getAllMessagesFromRoom(room: Room) {
-    const timeline = room.getLiveTimeline().getEvents();
-    const messages = timeline
-      .filter((event) => event.getType() === EventType.RoomMessage && !event.isRedacted())
-      .map(this.mapMatrixEventToMessage);
-    return messages;
+  private getAllMessagesFromRoom(room: Room): any[] {
+    const events = room
+      .getLiveTimeline()
+      .getEvents()
+      .reverse()
```

Comment:
I wonder if this messes up the internal state of the room. Maybe we need to "clone" the array before reversing it?

Reply:
Nice yeah to avoid potential side effects, I've cloned/copied the array before reversing it. Thanks!